api/handlers/dataexplorerhandler.py

```diff
@@ -450,6 +450,112 @@ def get_facets(self):
         aggs['by_session']['subject.age'] = age_node['subject.age']
         return {'facets': aggs}
 
+    def search_size(self, return_type):
+        body = {
+            "size": 0,
+            "aggs" : {
+                "count" : {
+                    "cardinality" : {
+                        "field" : return_type + "._id",
+                        "precision_threshold": 100000
+                    }
+                }              
+            }
+        }
+
+        size = config.es.search(
+            index='data_explorer',
+            doc_type='flywheel',
+            body=body)['aggregations']['count']['value']
+        size = int(size*1.02)
+        return size
+
+    def get_nodes(self):
+
+        return_type, filters, search_string = self._parse_request()
+        if return_type == 'file':
+            return self.get_file_nodes(return_type, filters, search_string)
+
+        size = self.search_size(return_type)
+        body = {
+            "size": 0,
+            "_source": ["session._id"],
```

Comment:
we'll want this to be based on return_type, I'm sure this is just a hard code you missed taking out.